### ISSUE TITLE
In-memory storage for benchmarking

### ISSUE THREAD

**Issue by haraldng** [logcabin/logcabin#238]
Hello,

I'm master student doing my thesis where I investigate in performance of distributed consensus algorithms. In the documentation it says that the in-memory storage cannot be bootstrapped. Is there any way to run a LogCabin cluster with in-memory storage for benchmarking?

**Comment by ongardie** [comment]
Hi @haraldng, I don't know if there's an easy way to do that. It's sort of an unusual configuration, since most people want their data to be durable. Here are a couple of ideas:
1. You could easily run with `Segmented` storage against a `tmpfs`.
2. You might also be able to bootstrap the cluster with a `Segmented` storage member, then add a bunch of `Memory` storage members, then remove the original `Segmented` storage member.

**Comment by haraldng** [comment]
I will try using tmpfs. Thank you for the advice!

**Comment by haraldng** [comment]
Hi again,

Is the `writeEx` function used in the [Benchmark example](https://github.com/logcabin/logcabin/blob/master/Examples/Benchmark.cc#L228) a blocking function? Blocking in terms of that it only returns to the client when a majority of Raft servers have committed that write operation?

**Comment by ongardie** [comment]
> Is the `writeEx` function used in the [Benchmark example](https://github.com/logcabin/logcabin/blob/master/Examples/Benchmark.cc#L228) a blocking function? Blocking in terms of that it only returns to the client when a majority of Raft servers have committed that write operation?

Yep. The client library is meant to be pretty hard to use unsafely.
